include/picongpu/fields/LaserPhysics.hpp

```diff
@@ -17,83 +17,196 @@
-        const DataSpace<simDim> halfSimSize(subGrid.getGlobalDomain().size / 2);
-
-        DataSpace<simDim> centeredOrigin(globalCellOffset - halfSimSize);
-        return LaserManipulator(elong,
-                                centeredOrigin,
-                                phase);
-    }
-
-private:
-
-    GridLayout<simDim> m_layout;
-};
-}
+        void operator()(uint32_t currentStep) const
+        {
+            /* initialize the laser not in the first cell is equal to a negative shift
+             * in time
+             */
+            constexpr float_X laserTimeShift = laserProfiles::Profile::Unitless::initPlaneY * CELL_HEIGHT / SPEED_OF_LIGHT;
+
+            const uint32_t numSlides = MovingWindow::getInstance().getSlideCounter(currentStep);
+
+            /* Disable laser if
+             * - init time of laser is over or
+             * - we have periodic boundaries in Y direction or
+             * - we already performed a slide
+             */
+            bool const laserNone = ( laserProfiles::Profile::Unitless::INIT_TIME == float_X(0.0) );
+            bool const laserInitTimeOver =
+                ( ( currentStep * DELTA_T  - laserTimeShift ) >= laserProfiles::Profile::Unitless::INIT_TIME );
+            bool const topBoundariesArePeriodic =
+                ( Environment<simDim>::get().GridController().getCommunicationMask( ).isSet( TOP ) );
+            bool const boxHasSlided = ( numSlides != 0 );
+
+            bool const disableLaser =
+                laserNone ||
+                laserInitTimeOver ||
+                topBoundariesArePeriodic ||
+                boxHasSlided;
+            if( !disableLaser )
+            {
+                PMACC_VERIFY_MSG(
+                    laserProfiles::Profile::Unitless::initPlaneY < static_cast<uint32_t>( Environment<simDim>::get().SubGrid().getLocalDomain().size.y() ),
+                    "initPlaneY must be located in the top GPU"
+                );
+
+                PMACC_CASSERT_MSG(
+                    __initPlaneY_needs_to_be_greate_than_the_top_absorber_cells_or_zero,
+                    laserProfiles::Profile::Unitless::initPlaneY > ABSORBER_CELLS[1][0] ||
+                    laserProfiles::Profile::Unitless::initPlaneY == 0 ||
+                    laserProfiles::Profile::Unitless::INIT_TIME == float_X(0.0) /* laser is disabled e.g. laserNone */
+                );
+
+                /* Calculate how many neighbors to the left we have
+                 * to initialize the laser in the E-Field
+                 *
+                 * Example: Yee needs one neighbor to perform dB = curlE
+                 *            -> initialize in y=0 plane
+                 *          A second order solver could need 2 neighbors left:
+                 *            -> initialize in y=0 and y=1 plane
+                 *
+                 * Question: Why do other codes initialize the B-Field instead?
```

Comment:
That's the same as is written there.

Reply:
> always first state what you do

I agree. The title of the comment block (L154) is:

> Calculate how many neighbors to the left we have to initialize the laser in the E-Field

Reply:
I also find this long comment rather confusing, but for a different reason.

To me the issue is with `Because our fields are defined on the lower cell side (C-Style ftw)`. Each Yee cell by being a cell and not a node has two values for each field component. So it's not that the fields are defined on the lower side, but the indexing of nodes and cells in picongpu is organized in such a way. Also both ways of organizing Yee cells (E on edges, B on faces or vice versa) are popular, so that might potentially add to reader's confusion.

Maybe it's useful (or it even exists already, but I am unaware) to have a picture in documentation which explains how components are organized in a Yee cell and also what indexing is used.

Reply:
Good point! We can add some in a follow up to the manual, I have them in my diploma thesis: https://zenodo.org/record/15924